app/importers/reading/somerville_mega_reading_importer.rb

```diff
@@ -51,6 +51,8 @@ def read_or_fetch_csv
     raise 'unfinished'
     # TODO read from google or mock and return `streaming_csv` that responds
     # to #each_with_index
+    sheetsFetcher = GoogleSheetsImporter.new
+    sheetsFetcher.getSheetsFromFolder("Folder Name") #TODO get folder based on district/school setup
```

Comment:
@kevinrobinson was this what you had in mind for the sheets importer? Rather than fill a directory with temporary csvs, this just streams them all at once so they can be imported here. The piece that might need to change is the location of the sheets. Right now, this class just fetches everything in the single folder it's passed. Depending on the district/school setup we might want different folders for different purposes, or add in more control over which sheets are imported from a big shared folder. I'm not certain here whether all shared sheets should be imported the same way. 

In any case, we have plenty of flexibility in how this is implemented. Let me know if this looks like the right idea.

Reply:
Yep, I think this is great!  👍

Here's some small API changes that will let us use this in more places:
1. Let's update this to take a `folder_id`, rather than searching by the text name of the folder, to avoid the chance of collisions or mismatches.
2. If we split this into two public methods we could use it for both bulk importing everything in a folder, but also for importing or testing single sheets as well.  Maybe something like `get_file_ids(folder_id)` and `get_spreadsheet_contents(spreadsheet_id)`?

Also, are there any changes we'd need to make for this to work with spreadsheets that have multiple tabs, like the "example 3 (somerville 2nd grade reading)"?  This looks like it might assume they all have the same shape and concat them into the same file.  If the fetching can return either a list like `[{tab_name, csv}]` or hash like `{tab_name => csv}` then we can use this for sheets that have different kinds of tabs as well.  It's my bad that example 3 has the same structure in each tab :)